test/sets.jl

```diff
@@ -810,6 +822,19 @@ end
     @test replace((NaN, 1.0), NaN=>0.0) === (0.0, 1.0)
     @test replace([1, missing], missing=>0) == [1, 0]
     @test replace((1, missing), missing=>0) === (1, 0)
+
+    # test replace and friends for AbstractDicts
+    d1 = ReplaceDict(Dict(1=>2, 3=>4))
+    d2 = replace(d1, (1=>2) => (1=>"a"))
+    @test d2 == ReplaceDict(Dict(1=>"a", 3=>4))
```

Comment:
this doesn't test that `d2` is a `ReplaceDict`:
```suggestion
    @test d2 == Dict(1=>"a", 3=>4)
    @test d2 isa ReplaceDict
```